client/scripts/controllers/MonitoringController.js

```diff
@@ -17,6 +17,10 @@ require('../app').controller('MonitoringController', /* @ngInject */function ($s
   controller.filter = angular.copy($stateParams)
   controller.offline = false
 
+  // transform threat filter to nomenclature object
+  controller.filter.threat = controller.filter.threat && db.nomenclatures.main_threats[controller.filter.threat]
```

Comment:
No need to check the filter value as `[][undefined] === undefined`

```suggestion
  controller.filter.threat = db.nomenclatures.main_threats[controller.filter.threat]
```

Reply:
Nomenclatures may not be loaded - better to wait on the `db.nomenclatures.main_threats.$promise` and then to convert